web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -204,8 +209,18 @@
             List<String> studyIds = new ArrayList<>();
             List<String> sampleIds = new ArrayList<>();
             studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
-            result = mutationService.getSampleCountInMultipleMolecularProfiles(molecularProfileService
-                .getFirstMutationProfileIds(studyIds, sampleIds), sampleIds, null, true);
+            
+            // getSampleCountInMultipleMolecularProfiles is counting the number of mutations within filtered samples
+            // But that does not guarantee they are driver mutations.
+            // I created method getSampleCountInMultipleMolecularProfilesByAnnotation in the MutationMapper.
+            // Please consider using that when updating the following code
+            List<AnnotationFilter> annotationCancer;
+            AnnotationFilter annotationCase = new AnnotationFilter();
+            annotationCase.setPath("oncokb.oncogenicity");
+            annotationCase.setValues(new List<String>("YES", "NO", "YES,NO"));
```

Comment:
Here is the same logic of the work you did in filterByDriverMutation. AnnotationFilter should be generated based on  withOncoKBDriverMutationData. If Yes, the value will be converted  to proper oncogenicity value.